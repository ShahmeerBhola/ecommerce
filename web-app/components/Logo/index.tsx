import React from 'react';
import Link from 'next/link';
import config from '../../config';
import { constructUrl } from '../../utils';

type Props = {
  renderLink?: boolean;
};

const Logo: React.FunctionComponent<Props> = ({ renderLink = true }) => {
  const Image = (): JSX.Element => (
    <img className="mx-auto" src="/images/logo.png" alt={config.siteName} width={140} height={70} />
  );

  if (renderLink) {
    return (
      <Link href={constructUrl({ page: '' })}>
        <a>
          <Image />
        </a>
      </Link>
    );
  }
  return <Image />;
};

export default Logo;
