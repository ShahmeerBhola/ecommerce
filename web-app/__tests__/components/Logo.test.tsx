import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../../components/Logo';

describe('Logo component', () => {
  test('it renders properly with renderLink set to true', () => {
    const tree = renderer.create(<Logo renderLink />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with renderLink set to false', () => {
    const tree = renderer.create(<Logo renderLink={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
