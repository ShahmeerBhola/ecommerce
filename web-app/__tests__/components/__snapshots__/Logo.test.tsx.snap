// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component it renders properly with renderLink set to false 1`] = `
<img
  alt="Standout Specialties"
  className="mx-auto"
  height={79}
  src="https://storage.googleapis.com/standout-specialties-public-assets/logo.png"
  width={161}
/>
`;

exports[`Logo component it renders properly with renderLink set to true 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <img
    alt="Standout Specialties"
    className="mx-auto"
    height={79}
    src="https://storage.googleapis.com/standout-specialties-public-assets/logo.png"
    width={161}
  />
</a>
`;
